latency-tests-analysis/scripts/generate_latency_report.py

```diff
@@ -4,42 +4,84 @@
 import matplotlib.pyplot as plt
 import textwrap
 import numpy as np
+import subprocess
 
 GREEN_COLOR = "#90EE90"
 RED_COLOR = "#F08080"
 
-def compute_latency(vm,output):
-    vm_path = f"{output}/ts_{vm}.txt"
-    pub_path = f"{output}/ts_sv_publisher.txt"
-    latencies = np.array([])
-    with open(vm_path, buffering=10**4) as vm_file, open(pub_path, buffering=10**4) as pub_file:
-        vm_lines = vm_file.readlines()
-        pub_lines = pub_file.readlines()
-        for vm_line, pub_line in zip(vm_lines, pub_lines):
-            iteration_vm, stream_vm, count_vm, ts_vm = vm_line.split(':')
-            iteration_pub, stream_pub, count_pub, ts_pub = pub_line.split(':')
-            if (iteration_vm, stream_vm, count_vm) == (iteration_pub, stream_pub, count_pub):
-                diff = int(ts_vm) - int(ts_pub)
-                latencies = np.append(latencies, diff)
+def count_line(filename):
+    out = subprocess.Popen(['wc', '-l', filename],
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT
+                         ).communicate()[0]
+    return int(out.partition(b' ')[0])
+
+def compute_latency(sub_sv_path, pub_sv_path):
+    latencies = np.zeros(2000)
+
+    sub_file = open(sub_sv_path, 'r')
+    pub_file = open(pub_sv_path, 'r')
+
+    line_count = 0
+    max_line = max(count_line(sub_sv_path), count_line(pub_sv_path))
+    while line_count < max_line:
+        line_count += 1
+        pub_line = pub_file.readline()
+        sub_line = sub_file.readline()
+        pub_it, pub_stream, pub_index, pub_ts = pub_line.split(':')
+        sub_it, sub_stream, sub_index, sub_ts = sub_line.split(':')
+        if((pub_it, pub_stream, pub_index) == (sub_it, sub_stream, sub_index)):
+            lat = int(sub_ts) - int(pub_ts)
+            if lat < 2001:
+                latencies[lat] += 1
+            else:
+                return -1
```

Comment:
I think it would be more clear to raise the exception here directy. 
Instead of passing "-1" as a return value.